code_search/src/code_search/t2t/function_docstring.py

```diff
@@ -1,32 +1,37 @@
 """Github function/text similatrity problems."""
 import csv
-import os
-from cStringIO import StringIO
+from six import StringIO
 from tensor2tensor.data_generators import generator_utils
-from tensor2tensor.data_generators import translate
+from tensor2tensor.data_generators import text_problems
 from tensor2tensor.utils import metrics
 from tensor2tensor.utils import registry
+import tensorflow as tf
 
 
-##
-# These URLs are only for fallback purposes in case the specified
-# `data_dir` does not contain the data. However, note that the data
-# files must have the same naming pattern.
-# TODO: The memory is exploding, need to fix this.
-#
-_DATA_BASE_URL = 'gs://kubeflow-examples/t2t-code-search/data'
-_GITHUB_FUNCTION_DOCSTRING_FILES = [
-    'pairs-0000{}-of-00010.csv'.format(i)
-    for i in range(1)
-]
+@registry.register_problem
+class GithubFunctionDocstring(text_problems.Text2TextProblem):
+  """Function and Docstring similarity Problem.
 
+  This problem contains the data consisting of function
+  and docstring pairs as CSV files. The files are structured
+  such that they contain two columns without headers containing
+  the docstring tokens and function tokens. The delimiter is
+  ",".
+  """
 
-@registry.register_problem
-class GithubFunctionDocstring(translate.TranslateProblem):
-  # pylint: disable=abstract-method
+  @property
+  def base_url(self):
+    return "gs://kubeflow-examples/t2t-code-search/raw_data"
 
-  """This class defines the problem of finding similarity between Python
-  function and docstring"""
+  @property
+  def pair_files_list(self):
+    return [
+        [
+            "{}/func-doc-pairs-000{:02}-of-00100.csv".format(self.base_url, i),
```

Comment:
Why are file paths being hard coded here?

Reply:
The semantics of T2T are such that all the data and files are deterministically defined. I had earlier made them dynamic but Lukasz suggested to not do that. Instead it is easy to just derive from this class and override the property to change the file names.

Reply:
Please add a comment explaining that.
Couldn't we just parameterize the derived class?

Reply:
Unfortunately, the T2T interface does not allow parametrization this way as it considers the dataset as an immutable. However, I will take this up with the T2T team if they have a better way of dynamically allowing this via CLI.